x-pack/legacy/plugins/snapshot_restore/server/lib/get_managed_repository_name.ts

```diff
@@ -0,0 +1,26 @@
+/*
+ * Copyright Elasticsearch B.V. and/or licensed to Elasticsearch B.V. under one
+ * or more contributor license agreements. Licensed under the Elastic License;
+ * you may not use this file except in compliance with the Elastic License.
+ */
+
+export const getManagedRepositoryName = async (
+  callWithInternalUser: any
+): Promise<string | undefined> => {
+  try {
+    const { persistent, transient, defaults } = await callWithInternalUser('cluster.getSettings', {
```

Comment:
Can we add a comment here to help people connect the dots with the Cloud-based business logic we're supporting? Here's my suggestion.

```
// Cloud has its own system for managing snapshots and we want to make
// this clear when Snapshot Restore is used in a Cloud deployment.
```